File: BOJ/Python/5532-방학숙제.py
inputList = []
for i in range(5):
    inputList.append(int(input()))
def up(num):
    if num > int(num):
        return int(num)+1
    return int(num)
maxDay = max(up(inputList[1]/inputList[3]), up(inputList[2]/inputList[4]))
print(inputList[0]-maxDay)

File: BOJ/C++/5532-방학 숙제.cpp
#include <cstdio>

int l, a, b, c, d;

int max(int a, int b) {
    return (a>b)?a:b;
}

int main() {
    scanf("%d %d %d %d %d", &l, &a, &b, &c, &d);
    printf("%d", l - max(a/c + (a%c > 0), b/d + (b%d > 0)));
}